lib/facility.rb

```diff
@@ -1,14 +1,62 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicle, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_information)
+    @name = facility_information[:name]
+    @address = facility_information[:address]
+    @phone = facility_information[:phone]
     @services = []
+    @registered_vehicle = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
+      @registered_vehicle << vehicle
+      vehicle.car_registration_date
+      collect_fees(vehicle)
+      @registration_date = vehicle.car_registration_date
```

Comment:
Don't need to create an instance variable here for @registration_date, calling just vehicle.car_registration_date will accomplish what you need. In terms of naming, remember that method names should be verbs, so maybe calling this method something like vehicle.set_registration_date would have been more indicative of what it does. 
Overall, great job using all these helper methods with this register_vehicle method.!